src/components/Send/SourceJarSelector.tsx

```diff
@@ -34,6 +57,76 @@ export const SourceJarSelector = ({
     )
   }, [walletInfo])
 
+  useEffect(() => {
+    if (showUtxos.jarIndex && walletInfo?.utxosByJar) {
+      const data = Object.entries(walletInfo.utxosByJar).find(([key]) => key === showUtxos.jarIndex)
+      const utxos: any = data ? data[1] : []
+
+      const frozenUtxoList = utxos
+        .filter((utxo: any) => utxo.frozen)
+        .map((utxo: any) => ({ ...utxo, id: utxo.utxo, checked: false }))
+      const unFrozenUtxosList = utxos
+        .filter((utxo: any) => !utxo.frozen)
+        .map((utxo: any) => ({ ...utxo, id: utxo.utxo, checked: true }))
+
+      setFrozenUtxos(frozenUtxoList)
+      setUnFrozenUtxos(unFrozenUtxosList)
+    }
+  }, [walletInfo, showUtxos.jarIndex, t])
+
+  useEffect(() => {
+    if (frozenUtxos.length === 0 && unFrozenUtxos.length === 0) {
+      return
+    }
+    const frozenUtxosToUpdate = frozenUtxos.filter((utxo: Utxo) => utxo.checked && !utxo.locktime)
+    const timeLockedUtxo = frozenUtxos.find((utxo: Utxo) => utxo.checked && utxo.locktime)
+    const allUnFrozenUnchecked = unFrozenUtxos.every((utxo: Utxo) => !utxo.checked)
+
+    if (frozenUtxos.length > 0 && timeLockedUtxo) {
+      setAlert({ variant: 'danger', message: `${t('show_utxos.alert_for_time_locked')} ${timeLockedUtxo.locktime}` })
+    } else if (
+      (frozenUtxos.length > 0 || unFrozenUtxos.length > 0) &&
+      allUnFrozenUnchecked &&
+      frozenUtxosToUpdate.length === 0
+    ) {
+      setAlert({ variant: 'warning', message: t('show_utxos.alert_for_unfreeze_utxos'), dismissible: true })
+    } else {
+      setAlert(undefined)
+    }
+  }, [frozenUtxos, unFrozenUtxos, t, setAlert])
+
+  const handleUtxosFrozenState = useCallback(async () => {
+    const abortCtrl = new AbortController()
+    const frozenUtxosToUpdate = frozenUtxos
+      .filter((utxo) => utxo.checked && !utxo.locktime)
+      .map((utxo) => ({ utxo: utxo.utxo, freeze: false }))
+    const unFrozenUtxosToUpdate = unFrozenUtxos
+      .filter((utxo) => !utxo.checked)
+      .map((utxo) => ({ utxo: utxo.utxo, freeze: true }))
+
+    try {
+      const res = await Promise.all([
+        ...frozenUtxosToUpdate.map((utxo) => Api.postFreeze({ ...wallet, signal: abortCtrl.signal }, utxo)),
+        ...unFrozenUtxosToUpdate.map((utxo) => Api.postFreeze({ ...wallet, signal: abortCtrl.signal }, utxo)),
+      ])
+
+      if (res.length !== 0) {
+        setIsUtxosLoading(true)
+        await reloadCurrentWalletInfo.reloadUtxos({ signal: abortCtrl.signal })
+        setIsUtxosLoading(false)
```

Comment:
If the call `reloadUtxos` fails, `setIsUtxosLoading` is not reset.
Can/Should this be handled in a `finally` callback?

Reply:
Ooh, how did I miss that, sry.